src/blitter/32bpp_anim.cpp

```diff
@@ -475,12 +475,12 @@ void Blitter_32bppAnim::ScrollBuffer(void *video, int &left, int &top, int &widt
 			src -= scroll_x;
 		}
 
-		/* the y-displacement may be 0 therefore we have to use memmove,
+		/* the y-displacement may be 0 therefore we have to use std::move_backward,
 		 * because source and destination may overlap */
 		uint tw = width + (scroll_x >= 0 ? -scroll_x : scroll_x);
 		uint th = height + scroll_y;
 		for (; th > 0; th--) {
-			memmove(dst, src, tw * sizeof(uint16_t));
+			std::move_backward(src, src + tw, dst + tw);
```

Comment:
If `scroll_y == 0` then `scroll_x` decides whether you need `std::move` or `std::move_backwards`.
But putting that logic here is too complicated IMO. Maybe `core/` should gain a `move_overlapping` method, which automatically decised between `std::move` and `std::move_backwards`, just like `memmove` did.

Reply:
Given the blitters are the only ones needing it, and the move code being the same for this bit for all of them... I made a new function so all blitters can call that. It will make the choice between the forward and backward move.